tool/doc_packages.dart

```diff
@@ -124,17 +124,20 @@ Future<List<String>> _packageUrls(int page) {
   return http
       .get(Uri.parse('https://pub.dartlang.org/packages.json?page=$page'))
       .then((response) {
-    return List<String>.from(json.decode(response.body)['packages']);
+    var decodedJson = json.decode(response.body) as Map;
+    return List.of(decodedJson['packages']);
   });
 }
 
 Future<List<PackageInfo>> _getPackageInfos(List<String> packageUrls) {
   var futures = packageUrls.map((String p) {
     return http.get(Uri.parse(p)).then((response) {
-      var decodedJson = json.decode(response.body);
+      var decodedJson = json.decode(response.body) as Map;
       String name = decodedJson['name'];
-      var versions = List<Version>.from(
-          decodedJson['versions'].map((v) => Version.parse(v)));
+      var versions = [
```

Comment:
Assuming inference works properly for the list type here?

Reply:
Right, `Version.parse` returns a Version, and this is the only element (repeated) in the list, so inference will find Version.